core/src/test/java/io/onetable/iceberg/ITIcebergSourceClient.java

```diff
@@ -297,6 +300,50 @@ public void testExpireSnapshots(boolean isPartitioned) throws InterruptedExcepti
     }
   }
 
+  @SneakyThrows
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testForIncrementalSyncSafetyCheck(boolean shouldExpireSnapshots) {
+    String tableName = getTableName();
+    try (TestIcebergTable testIcebergTable =
+        TestIcebergTable.forStandardSchemaAndPartitioning(
+            tableName, "level", tempDir, hadoopConf)) {
+      // Insert 50 rows to INFO partition.
+      List<Record> commit1Rows = testIcebergTable.insertRecordsForPartition(50, "INFO");
+      Long timestamp1 = testIcebergTable.getLastCommitTimestamp();
+      PerTableConfig tableConfig =
+          PerTableConfig.builder()
+              .tableName(testIcebergTable.getTableName())
+              .tableBasePath(testIcebergTable.getBasePath())
+              .tableDataPath(testIcebergTable.getDataPath())
+              .targetTableFormats(Arrays.asList(TableFormat.HUDI, TableFormat.DELTA))
+              .build();
+
+      // Upsert all rows inserted before, so all files are replaced.
+      testIcebergTable.upsertRows(commit1Rows.subList(0, 50));
+      long snapshotIdAfterCommit2 = testIcebergTable.getLatestSnapshot().snapshotId();
+      Thread.sleep(5 * 1000);
```

Comment:
why do you need a sleep here?

Reply:
I was playing with instants before and to round up I used this, no longer needed.